lib/vault/client.rb

```diff
@@ -296,6 +296,8 @@ def request(verb, path, data = {}, headers = {})
           request(verb, response[LOCATION_HEADER], data, headers)
         when Net::HTTPSuccess
           success(response)
+        when Net::HTTPPreconditionFailed
+          raise MissingRequiredStateError.new
```

Comment:
Does this belong in the `error()` method called in the `else` here, do you think? Seems like it as other `Net` exceptions are handled there as well. 

https://github.com/hashicorp/vault-ruby/blob/master/lib/vault/client.rb#L381-L415